projects/storefrontlib/layout/a11y/keyboard-focus/skip-focus.directive.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+
+export interface SkipFocusConfig {
+  isEnabled: boolean;
+  skipSelectors?: string[];
+}
+
+@Directive({
+  selector: '[cxSkipFocus]',
+})
+export class SkipFocusDirective implements OnChanges {
+  @Input('cxSkipFocus') config: SkipFocusConfig = { isEnabled: false };
+
+  constructor(
+    protected elementRef: ElementRef,
+    protected winRef: WindowRef
+  ) {}
+
+  public ngOnChanges(_changes: SimpleChanges): void {
```

Comment:
I think using `changes` could be a heavy-handed way to handle this issue. It is going to be present on every storefront component which contains the entire application. Meaning it will be a constant performance impact on every page. I've left a comment below about using an observable make changes to the directive. Maybe this would let us not need to watch `onChanges`?